### ISSUE TITLE
Regression in latest optimizations cause end of decompressed files to be cut off

### ISSUE THREAD

**Issue by Col-E** [Col-E/LL-Java-Zip#14]
Example:
```
196,415 // LFH compressed length
196,385 // LFH decompressed length
196,375 // literal data, before decompression
196,345 // literal data, after decompression
```
END entry starts at `196,363` which is beyond the decompressed data length

**Comment by Col-E** [comment]
Should reinvestigate if this was related to #16 or perhaps some similar sort of bug to it that hasn't been noticed yet.

**Comment by Col-E** [comment]
Still happens :(

**Comment by Col-E** [comment]
Actually, this does not seem related to compression or optimization, but a specific issue with the sample and how lljzip is used with it.

JVM strategy was used to read both the primary input, and the contents of a jar-in-jar. However, the JVM strategy realistically should not be used for embedded content since inner resources will typically be read with `ZipInputStream`. Thus, the forward scan/naive stategies ought to be used.

That being said, using those still led to some issues.

**Comment by Col-E** [comment]
Through a number of fixes in #20 and #19 this has been resolved.
